<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.31">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Søren Jørgensen">
<meta name="dcterms.date" content="2025-05-24">

<title>master_workflow – hic-compartment-borders</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./reports/example.html" rel="next">
<link href="./master_workflow.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-e1a5c8363afafaef2c763b6775fbf3ca.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-02d7654a0ad9d4caed0fef654f8ad003.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>


</head>

<body class="nav-sidebar floating quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./master_workflow.html">Workflows</a></li><li class="breadcrumb-item"><a href="./workflow.html"><span class="chapter-title">master_workflow</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">hic-compartment-borders</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/munch-group/hic-compartment-borders" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
    <a href="./hic-compartment-borders.pdf" title="Download PDF" class="quarto-navigation-tool px-1" aria-label="Download PDF"><i class="bi bi-file-pdf"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preface</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false">
 <span class="menu-text">Manuscript</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./manuscript/index.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Title of manuscript</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false">
 <span class="menu-text">Thesis</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./thesis/index.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Title of my thesis</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false">
 <span class="menu-text">Slides</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./slides/index.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Example</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false">
 <span class="menu-text">Notebooks</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./notebooks/01_preliminary_bayes.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Probabilistic Modeling of A/B Compartments from E1</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./notebooks/02_prep_e1_values.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Preparing the E1 values from the raw HIC data, but mapped to the new reference T2T-MMUv1</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true">
 <span class="menu-text">Workflows</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./master_workflow.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">GWF workflow</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./workflow.html" class="sidebar-item-text sidebar-link active"><span class="chapter-title">master_workflow</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" role="navigation" aria-expanded="false">
 <span class="menu-text">Reports</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./reports/example.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Markdown in Quarto</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul class="collapse">
  <li><a href="#index-the-reference-genome" id="toc-index-the-reference-genome" class="nav-link active" data-scroll-target="#index-the-reference-genome">Index the reference genome</a></li>
  </ul>
<div class="toc-actions"><ul class="collapse"><li><a href="https://github.com/munch-group/hic-compartment-borders/edit/main/workflow.py" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/munch-group/hic-compartment-borders/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li><li><a href="https://github.com/munch-group/hic-compartment-borders/blob/main/workflow.py" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./master_workflow.html">Workflows</a></li><li class="breadcrumb-item"><a href="./workflow.html"><span class="chapter-title">master_workflow</span></a></li></ol></nav>
<div class="quarto-title">
<h1 class="title"><span class="chapter-title">master_workflow</span></h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Søren Jørgensen </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">May 24, 2025</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<p>from gwf import * import os import subprocess import os.path as op import pandas as pd from cooler.fileops import list_coolers from pprintpp import pprint as pp</p>
<section id="section" class="level38">
<p class="heading"></p>
<p><code>gwf</code> workflow for the full pipeline from downloading raw reads to compartmentalization analysis. We are using <code>bwa mem</code> for mapping.</p>
<p>How to run: conda activate hic gwf -f master_workflow.py status</p>
<p>Workflow: 1a bwa_index : Index the reference genome with <code>bwa index</code> 1b sam_index : Index the fasta again with <code>samtools faidx</code> 1c download_reads : Download raw reads from SRA <code>sra-downloader</code> with SRR IDs (paired-end reads) 2 bwa_map : Map reads (PE) to the reference with <code>bwa mem</code> 3 pair_sort : Pair the merged alignments from mate-pair sequencing with <code>pairtools parse</code>, then sort with <code>pairtools sort</code> 4 dedup : Deduplicate the sorted pairs with <code>pairtools dedup</code> 5 make_pairs_cool : Create a .cool file from the deduplicated pairs with <code>cooler cload pairs</code></p>
</section>
<section id="section-1" class="level38">
<p class="heading"></p>
<p>Create a workflow object gwf = Workflow(defaults={‘nodes’: 1, ‘queue’:“normal”, ‘account’:“hic-spermatogenesis”})</p>
<section id="section-2" class="level44">
<p class="heading"></p>
</section>
</section>
<section id="templates" class="level14">
<p class="heading">Templates</p>
<section id="section-3" class="level44">
<p class="heading"></p>
<p>def download_index_reference(name, refdir): ““” Download the reference genome from NCBI, and index it for bwa ““” out_dir = op.join(refdir, name) inputs = [] outputs = [f”{out_dir}/index/bwa/{name}.fa”, f”{out_dir}/index/bwa/{name}.fa.amb”, f”{out_dir}/index/bwa/{name}.fa.ann”, f”{out_dir}/index/bwa/{name}.fa.bwt”, f”{out_dir}/index/bwa/{name}.fa.pac”, f”{out_dir}/index/bwa/{name}.fa.sa”, f”{out_dir}/assembly_report.txt”, f”{out_dir}/{name}.fa.sizes”, f”{out_dir}/{name}.fa.fai”, f”{out_dir}/{name}.gaps.bed”, ] options = { ‘memory’: ‘16g’, ‘walltime’: ‘01:00:00’, ‘cores’: 8, } spec = f”“” source $(conda info –base)/etc/profile.d/conda.sh conda activate hic genomepy plugin enable bwa &amp;&amp;<br>
genomepy install {name} –provider ncbi –genomes_dir {refdir} –threads 8 ““” return AnonymousTarget(inputs=inputs, outputs=outputs, options=options, spec=spec)</p>
<p>def download_reads(sra_downloader, srr_id, read_dir): “““Download raw reads from SRA”“” inputs = [sra_downloader] outputs = [f”{read_dir}/{srr_id}_1.fastq.gz”, f”{read_dir}/{srr_id}<em>2.fastq.gz”, f”{read_dir}/dl</em>{srr_id}.done”] options = {‘cores’:16, ‘memory’: “4g”, ‘walltime’:“06:00:00”} spec = f”“” singularity run {sra_downloader} {srr_id} –save-dir {read_dir} –cores 16 &amp;&amp;<br>
touch {read_dir}/dl_{srr_id}.done ““” return AnonymousTarget(inputs=inputs, outputs=outputs, options=options, spec=spec)</p>
<p>def bwa_index(ref_genome): “““Template for indexing the reference genome with bwa”“” inputs = [ref_genome] outputs = [f”{ref_genome}.amb”, f”{ref_genome}.ann”, f”{ref_genome}.bwt”, f”{ref_genome}.pac”, f”{ref_genome}.sa”] options = {‘cores’:1, ‘memory’: “8g”, ‘walltime’:“02:00:00”} spec = f’’’ source $(conda info –base)/etc/profile.d/conda.sh conda activate hic bwa index -p {ref_genome} -a bwtsw {ref_genome} ’’’ return AnonymousTarget(inputs=inputs, outputs=outputs, options=options, spec=spec)</p>
<p>def sam_index(ref_genome): “““Creating a fasta index. <code>bwa mem</code> also needs a fasta index generated by samtools”“” inputs = [ref_genome] outputs = [f”{ref_genome}.fai”] options = {‘cores’:1, ‘memory’:“1g”, ‘walltime’:“00:20:00”} spec=f”“” source $(conda info –base)/etc/profile.d/conda.sh conda activate hic samtools faidx {ref_genome} ““” return AnonymousTarget(inputs=inputs, outputs=outputs, options=options, spec=spec)</p>
<p>def bwa_map(ref_genome, mate_1, mate_2, download_ok, out_bam): ““” Template for mapping reads to a reference genome using <code>bwa</code> and <code>samtools</code>. NB! Here, we map the mates together, as bwa states it is no problem for Hi-C reads. ““” threads = 32 inputs = [f”{ref_genome}.amb”, f”{ref_genome}.ann”, f”{ref_genome}.bwt”, f”{ref_genome}.pac”, f”{ref_genome}.sa”, #f”{ref_genome}.fai”, mate_1, mate_2, download_ok] outputs = [out_bam] options = {‘cores’:threads, ‘memory’: “32g”, ‘walltime’:“24:00:00”} spec = f”“” source $(conda info –base)/etc/profile.d/conda.sh conda activate hic bwa mem -t {threads} -SP {ref_genome} {mate_1} {mate_2} &gt; {out_bam} ““” return AnonymousTarget(inputs=inputs, outputs=outputs, options=options, spec=spec)</p>
<p>Updated version uses –walks-policy 5unique in stead of mask def pair_sort_alignments(chromsizes, bam_merged, sorted_pairs): “““Convert the paired-end alignments to .pairs with <code>pairtools parse</code>, then sort”“” inputs = [bam_merged] outputs = [f”{bam_merged}_parsed.stats”, sorted_pairs] options = {‘cores’:32, ‘memory’:“16g”, ‘walltime’:“10:00:00”} spec=f”“” source $(conda info –base)/etc/profile.d/conda.sh conda activate hic pairtools parse<br>
-c {chromsizes}<br>
–drop-sam –drop-seq<br>
–output-stats {bam_merged}_parsed.stats<br>
–add-columns mapq<br>
–assembly rheMac10<br>
–walks-policy 5unique<br>
{bam_merged} |<br>
pairtools sort -o {sorted_pairs} ““” return AnonymousTarget(inputs=inputs, outputs=outputs, options=options, spec=spec)</p>
<p>def dedup(sorted_pairs, chromsizes): “““Deduplicate the sorted pairs with <code>pairtools dedup</code>”“” pairs_prefix = sorted_pairs.split(“.sorted”)[0] inputs = [sorted_pairs] outputs = [f”{pairs_prefix}.nodups.pairs.gz”, f”{pairs_prefix}.unmapped.pairs.gz”, f”{pairs_prefix}.dups.pairs.gz”, f”{pairs_prefix}.dedup.stats”, f”{pairs_prefix}.dedup.done”] options = {‘cores’:12, ‘memory’: “4g”, ‘walltime’: “03:30:00”} spec = f”“” source $(conda info –base)/etc/profile.d/conda.sh conda activate hic pairtools dedup<br>
–max-mismatch 3<br>
–mark-dups<br>
–chunksize 100000<br>
–chrom-subset {chromsizes}<br>
–output {pairs_prefix}.nodups.pairs.gz<br>
–output-unmapped {pairs_prefix}.unmapped.pairs.gz<br>
–output-stats {pairs_prefix}.dedup.stats<br>
–output-dups {pairs_prefix}.dups.pairs.gz<br>
{sorted_pairs} &amp;&amp;<br>
touch {pairs_prefix}.dedup.done ““” return AnonymousTarget(inputs=inputs, outputs=outputs, options=options, spec=spec)</p>
<p>Updated version uses pairtools select to filter out low quality reads def make_pairs_cool(chromsizes, pairs, cool_out): “““Create coolers from pairs with <code>cooler cload pairs</code>”“” base = os.path.basename(pairs).split(“.pairs.gz”)[0] inputs = [pairs] outputs = [cool_out] options = {‘cores’:1, ‘memory’:“8g”, ‘walltime’:“06:30:00”} spec = f”“” source $(conda info –base)/etc/profile.d/conda.sh conda activate hic pairtools select “(mapq1&gt;=30) and (mapq2&gt;=30)” {pairs} |<br>
cooler cload pairs<br>
-c1 2 -p1 3 -c2 4 -p2 5<br>
–assembly ‘T2T-MMU8v1’<br>
–chunksize 200000<br>
{chromsizes}:1000<br>
-<br>
{cool_out} ““” return AnonymousTarget(inputs=inputs, outputs=outputs, options=options, spec=spec)</p>
<p>def merge_zoomify_balance(cooler_list, merged, mcool): “““Merge a given list of coolers with <code>cooler merge</code> and then zoomify the merged cooler with <code>cooler zoomify</code>”“” cooler_list_unix_str = ” “.join(cooler_list) inputs = [cooler_list] outputs = [merged, mcool] options = {‘cores’:16, ‘memory’:”32g”, ‘walltime’:“08:00:00”} spec = f”“” source $(conda info –base)/etc/profile.d/conda.sh conda activate hic cooler merge -c 50000000 {merged} {cooler_list_unix_str} &amp;&amp;<br>
cooler zoomify –nproc 16<br>
–resolutions 1000,5000,10000,50000,100000<br>
–balance<br>
–balance-args ‘–nproc 16’<br>
-o {mcool}<br>
{merged} ““” return AnonymousTarget(inputs=inputs, outputs=outputs, options=options, spec=spec)</p>
<p>DRAFT: Not implemented as a target yet def balance_cooler_cis(cool_in, cool_out): “““Balance a cooler with <code>cooler balance</code>”“” mcool = cool_in.split(“::”)[0] inputs = [mcool] outputs = [cool_out] options = {‘cores’:8, ‘memory’:“32g”, ‘walltime’:“03:00:00”} spec = f”“” source $(conda info –base)/etc/profile.d/conda.sh conda activate hic cooler balance -p 32 –cis-only –name cis_weights {cool_in} &amp;&amp;<br>
touch {cool_out} ““” return AnonymousTarget(inputs=inputs, outputs=outputs, options=options, spec=spec)</p>
<section id="section-4" class="level47">
<p class="heading"></p>
<p>Set up the folder structure for the workflow # ###############################################</p>
<p># Define our working dir main_dir = op.dirname(<strong>file</strong>) msc_dir = op.join(main_dir, “../hic-spermatogenesis/”)</p>
<p>Define the reads directory (download reads here) reads_dir = op.join(“steps/macaque_raw/downloaded/”)</p>
<p>Check to find SRA-downloader, else pull from docker sra_downloader = “steps/sra-downloader.sif”</p>
<p>if not op.exists(sra_downloader): command = [“singularity”, “pull”, sra_downloader, “docker://wwydmanski/sra-downloader”] try: result = subprocess.run(command, check=True, capture_output=True, text=True) print(result.stdout) print(result.stderr) except subprocess.CalledProcessError as e: print(e.stdout) print(e.stderr)</p>
<p>Define subdirs for tissue type, strip_whitespace and make lowercase sra_runtable = pd.read_csv(“data/SraRunTable.txt”) reads_subdirs = set(sra_runtable[“source_name”])</p>
<p>Make a dict mapping the tissue type [‘source_name’] to the SRR IDs [‘Run’] tissue_dict = {tissue: sra_runtable[sra_runtable[“source_name”]==tissue][“Run”].tolist() for tissue in reads_subdirs}</p>
<p>Define the output dirs for files bam_dir = op.join(main_dir,“steps/bamfiles”) pair_dir = op.join(main_dir, “steps/pairs”)<br>
cool_dir = op.join(main_dir, “steps/cool”)</p>
</section>
</section>
<section id="section-5" class="level44">
<p class="heading"></p>
</section>
</section>
<section id="targets" class="level14">
<p class="heading">Targets</p>
<section id="section-6" class="level44">
<p class="heading"></p>
<p>Download the reference genome ref_name = “T2T-MMU8v1.0” ref_dir = op.join(main_dir, “steps/ref”)</p>
<p>T0 = gwf.target_from_template( f”install_T2T_MMU8V1”, download_index_reference(ref_name, ref_dir) )</p>
<p>ref_genome = T0.outputs[0] ref_dir = op.join(ref_dir, ref_name) ## Update ref_dir chromsizes = op.join(ref_dir, f”{ref_name}.fa.sizes”)</p>
</section>
</section>
<section id="index-the-reference-genome" class="level1">
<h1>Index the reference genome</h1>
<p>T1a = gwf.target_from_template(f”bwa_index_{os.path.basename(ref_genome).split(‘.’)[0]}“, bwa_index(ref_genome=ref_genome)) T1b = gwf.target_from_template(f”sam_index_{os.path.basename(ref_genome).split(‘.’)[0]}“, sam_index(ref_genome=ref_genome))</p>
<p>T5out is a dict to store the cool files for each tissue type to be merged in the end T5out = {}</p>
<p>T1c target for each ID in the SRA runtable for id in sra_runtable[“Run”]:</p>
<pre><code># Do some stuff before downloading the reads
# Get the tissue type for the ID, and assign a subdirectory for the reads
sub_dir = None
for cell_type, ids in tissue_dict.items():
    if id in ids:
        sub_dir = cell_type.lower().replace(" ", "_")
        break
if sub_dir is None:
    raise ValueError(f"ID {id} not found in any tissue type")

fastq_dir = op.join(reads_dir, sub_dir)

# Create subdirs if they don't exist
if not op.exists(fastq_dir):
    os.makedirs(fastq_dir)

# T1c: Download SRA ID
T1c = gwf.target_from_template(f"dl_{id}",
                              download_reads(
                                    sra_downloader = sra_downloader,
                                    srr_id = id,
                                    read_dir = fastq_dir
                                ))


# T2: Map the reads to the reference genome
bam_wdir = op.join(bam_dir, sub_dir)
out_bam = op.join(bam_wdir, f"{id}.bam")

if not op.exists(bam_wdir):
    os.makedirs(bam_wdir)

T2 = gwf.target_from_template(f"bwa_map_{id}", 
                              bwa_map(ref_genome=ref_genome, 
                                      mate_1=T1c.outputs[0], mate_2=T1c.outputs[1], 
                                      download_ok=T1c.outputs[2],
                                      out_bam=out_bam))

# T3: pair and sort the alignments
pair_subdir = op.join(pair_dir, sub_dir)
sorted_pairs = op.join(pair_subdir, f"{id}.sorted.pairs.gz")

if not op.exists(pair_subdir):
    os.makedirs(pair_subdir)

T3 = gwf.target_from_template(f"parse_{id}",
                                pair_sort_alignments(chromsizes=chromsizes, 
                                                     bam_merged=T2.outputs[0], 
                                                     sorted_pairs=sorted_pairs))

# T4: Deduplicate the sorted pairs
# Dedup dir is the same as pair dir
# NB we are using the reduced chromsizes file (only 'real' chromosomes)
T4 = gwf.target_from_template(f"dedup_{id}",
                                dedup(sorted_pairs=T3.outputs[1],
                                        chromsizes=chromsizes))

# T5: Create a .cool file from the deduplicated pairs
cool_subdir = op.join(cool_dir, sub_dir)

if not op.exists(cool_subdir):
    os.makedirs(cool_subdir)

T4outpairs = [x for x in T4.outputs if x.endswith(".pairs.gz")]
if sub_dir not in T5out:
    T5out[sub_dir] = []



for T4out in T4outpairs:
    # Filter to be removed if we want to
    # Only make a cool file of the 'nodups' pairs
    if 'nodups' not in T4out:
        continue

    cool_name = f"{id}"
    cool_file = os.path.join(cool_subdir, f"{cool_name}.1000.cool")

    T5 = gwf.target_from_template(f"coolify_{cool_name}",
                                    make_pairs_cool(chromsizes, pairs=T4out, cool_out=cool_file))
    T5out[sub_dir].append(T5.outputs[0])
    </code></pre>
<p>Merge the coolers for each tissue type (group) T6out = {} for subdir,cool_list in T5out.items(): T6out[subdir] = gwf.target_from_template(f”merge_zoom_balance_{subdir}“, merge_zoomify_balance(cooler_list=cool_list, merged=op.join(cool_dir, subdir, f”{subdir}.merged.cool”), mcool=op.join(cool_dir, subdir, f”{subdir}.merged.mcool”))) pp(T6out[subdir].inputs[0])</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./master_workflow.html" class="pagination-link" aria-label="GWF workflow">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-title">GWF workflow</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./reports/example.html" class="pagination-link" aria-label="Markdown in Quarto">
        <span class="nav-page-text"><span class="chapter-title">Markdown in Quarto</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




<footer class="footer"><div class="nav-footer"><div class="nav-footer-center"><div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/munch-group/hic-compartment-borders/edit/main/workflow.py" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/munch-group/hic-compartment-borders/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li><li><a href="https://github.com/munch-group/hic-compartment-borders/blob/main/workflow.py" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></div></div></footer></body></html>